Sources/SwiftParser/Expressions.swift

```diff
@@ -917,16 +919,37 @@ extension Parser {
     /// Parse close quote.
     let closeQuote = self.parseStringLiteralQuote(
       at: openDelimiter != nil ? .trailingRaw : .trailing,
-      text: text[closeStart...]
+      text: text
     ) ?? RawTokenSyntax(missing: openQuote.tokenKind, arena: arena)
-    text = text.dropFirst(closeQuote.byteLength)
-
+    if !closeQuote.isMissing {
+      text = text.dropFirst(closeQuote.tokenText.count)
+    }
     /// Parse closing raw string delimiter if exist.
-    let closeDelimiter = self.parseStringLiteralDelimiter(at: .trailing, text: text)
+    let closeDelimiter: RawTokenSyntax? = {
+      if let closeDelimiter = self.parseStringLiteralDelimiter(
+        at: .trailing,
+        text: text
+      ) {
+        return closeDelimiter
+      }
+
+      if let openDelimiter = openDelimiter {
+        return RawTokenSyntax(
+          kind: .rawStringDelimiter,
+          text: openDelimiter.tokenText,
+          leadingTriviaPieces: [],
+          trailingTriviaPieces: [],
+          presence: .missing,
+          arena: arena
+        )
```

Comment:
You could use
```suggestion
        return RawTokenSyntax(
          missing: .rawStringDelimiter,
          text: openDelimiter.tokenText,
          arena: arena
        )
```

If you add `text: SyntaxText? = nil,` to `RawTokenText.init(missing:arena:)` and apply the following change in the initializer.
```diff
- text: kind.defaultText ?? "",
+ text: text ?? kind.defaultText ?? "",
```

I think I’ve got at least two branches locally that do this as well…

Reply:
Actually I looked for `RawTokenText(missing:text:arena:)` first time 😄.
I added `text:` and changed to use it.